### PR TITLE
generic WorkerComm.js to communicate between phoenix and web workers

### PR DESCRIPTION
**Pull request by abose**
- feat: create generic WorkerComm.js to communicate between phoenix and web workers
- feat: dynamic loading worker js with loadScriptInWorker API
- port find in files api to workercomm infrastructure

## Testing
* All unit/integ tests working/failing as expected.
* new unit tests added for WorkerComm

### CODE REVIEW

**Comment by phoenixide** [comment]
## Thank You for making this pull request.

**Comment by phoenixide** [comment]
## Thank You for making this pull request.

**Comment by phoenixide** [comment]
## Thank You for making this pull request.

File: test/spec/WorkerComm-worker.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://opensource.org/licenses/AGPL-3.0.
+ *
+ */
+
+/*global WorkerComm*/
+
+const urlParams = new URLSearchParams(location.search);
+importScripts(urlParams.get('eventDispatcherURL'));
```
[the remark this answers is not in the document]
Comment (abose) [reply test/spec/WorkerComm-worker.js:24]: used in tests

File: test/spec/WorkerComm-worker.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+importScripts(urlParams.get('workerCommUrl'));
```
[the remark this answers is not in the document]
Comment (abose) [reply test/spec/WorkerComm-worker.js:25]: used in tests

**Pull request merged**